### PR TITLE
feat: upgrade bn.js from 4.12 to 5.2

### PR DESCRIPTION
**Pull request by pradel**
## Description

At the moment @stacks/connect is using bn.js version 5.2 and the packages of this repo the version 4.12. This means that when we use both connect and stacks.js packages in our app, we will end up having 2 versions of bn.js in our bundle.

What I did was to upgrade bn.js dependency from 4.12 to the next major 5.2 so both connect and stacks.js use the same version.

stacks-wallet-web is already using 5.2 too https://github.com/blockstack/stacks-wallet-web/blob/main/package.json#L71

Related to issue #1120, I noticed that bn.js is included twice in my bundle.

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] API reference/documentation update
- [x] Other

## Does this introduce a breaking change?

All the tests are passing so not sure if this qualifies as a breaking change as users won't be impacted directly by this change.

## Are documentation updates required?

I don't think so.

## Checklist
- [X] Code is commented where needed
- [X] Unit test coverage for new or modified code paths
- [X] `npm run test` passes
- [ ] Changelog is updated
- [ ] Tag 1 of @yknl or @zone117x for review

### CODE REVIEW

File: packages/encryption/package.json [hunk 1]
```diff
@@ -31,11 +31,11 @@
   },
   "dependencies": {
     "@stacks/common": "^2.0.1",
-    "@types/bn.js": "^4.11.6",
+    "@types/bn.js": "^5.1.0",
     "@types/node": "^14.14.43",
     "bip39": "^3.0.2",
     "bitcoinjs-lib": "^5.2.0",
```
Comment (zone117x) [inline packages/encryption/package.json:37]: IIRC it was the `bitcoinjs` family of libs (this one and some of the `bip` libs) that were including `bn.js` v4. Have you tested that these dependencies don't end up requiring `bn.js` v4 to be bundled anyway?

Comment (pradel) [reply packages/encryption/package.json:37]: Other packages are depending of v4, my idea was to do the upgrade gradually, here is the list of deps still using it (might have missed some):

- @stacks/keychain
  - @blockstack/rpc-client 0.3.0-alpha.11
    - @blockstack/stacks-transactions 0.5.1
- Used by multiple packages
  - jsontokens 3.0.0
    - asn1.js 5.4.1
- @stacks/cli 2.0.1
  - blockstack 19.3.0
- Used by multiple packages
  - elliptic 6.4.5
- bitcoin-lib 5.2.0
  - tiny-secp256k1 1.1.6

The issue can be that some of them seem unmaintained, like https://github.com/indutny/asn1.js/pull/118 where the pr was opened almost 2 years ago.

Comment (zone117x) [reply packages/encryption/package.json:37]: I'd be more concerned with getting the `elliptic` package and bitcoinjs-lib packages upgraded to bn.js v5. Is there anything in bn.js v5 that is needed? And would be more feasible to change projects using v5 to use v4 instead?

Comment (pradel) [reply packages/encryption/package.json:37]: Looks like bitcoin-js lib is working on removing the dependency to bn.js as they are dropping tiny-secp256k1 https://github.com/bitcoinjs/bitcoinjs-lib/pull/1734. For elliptic it looks more challenging.
I have no idea if you are using v5 specific features on the other packages.

Comment (junderw) [reply packages/encryption/package.json:37]: We are only dropping bn.js from the main package, ecpair and bip32 will still use it until we can upgrade them to use WASM.

**Comment by diwakergupta** [comment]
Hi @pradel -- based on the comments, it doesn't look like this PR would actually help reduce the bundle size. I'm going to close it for now, but please reopen if you disagree. Meanwhile, we _are_ looking into all the different pain points with stacks.js, including bundle sizes, and hope to have a plan we can share with the community in the coming weeks. Thanks!

**Pull request closed without merging**